### PR TITLE
Add uWSGI support

### LINKED ISSUE #38: Feature: Support automatically configuring uWSGI pass
Hi there,

We're currently not seeing distributed tracing being automatically configured when using a [uwsgi_pass](http://nginx.org/en/docs/http/ngx_http_uwsgi_module.html). It'd be ideal if this just worked upon loading of the module (similar to how HTTP proxy passes function). Would something like that be feasible as a feature?

### PR DESCRIPTION
**Pull request by dmehala**
This PR add support for uWSGI. When `uwsgi_pass` directives are encountered, the module automatically propagate traces.

Content:
- Add uWSGI service for testing and as an example
- Add propagation tests for uwsgi directives
- Update example/README.md
- Ignore *.pyc

Resolves #38

### CODE REVIEW

File: test/services/uwsgi/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM alpine:3.15
+
+RUN mkdir /opt/app
+WORKDIR /opt/app
+
+RUN apk update && apk add python3 py3-pip python3-dev build-base linux-headers pcre-dev
+RUN pip3 install uwsgi flask ddtrace
```
Comment (dmehala) [inline test/services/uwsgi/Dockerfile:7]: Add `--no-cache-dir`

File: example/README.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ services:
 - `http` runs a NodeJS HTTP server that listens on port 8080.
 - `fastcgi` runs a NodeJS FastCGI server that listens on port 8080.
 - `grpc` runs a NodeJS gRPC server that listens on port 1337.
+- `uwsgi` runs a uWSGI server using the `uwsgi` protocol and listens on port 8080.
```
[the remark this answers is not in the document]
Comment (dmehala) [reply example/README.md]: Resolved in d475f16d67b2f57a20161586b2ddf417903afeaf

File: example/services/uwsgi/__pycache__/app.cpython-311.pyc
[the remark this answers is not in the document]
Comment (dmehala) [reply example/services/uwsgi/__pycache__/app.cpython-311.pyc:1]: Not at all. I guess it was before I added the new rule in `.gitignore`. Resolved in d475f16d67b2f57a20161586b2ddf417903afeaf

File: example/services/uwsgi/uwsgi.ini [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[uwsgi]
+module = wsgi:app
+master = true
+socket = :8080
+
+;; ddtrace required options
+enable-threads = 1
+lazy-apps = 1
+import=ddtrace.bootstrap.sitecustomize
```
[the remark this answers is not in the document]
Comment (dmehala) [reply example/services/uwsgi/uwsgi.ini:11]: I added more informations. As a non native english speaker, any suggestion is fine :)

File: src/datadog_directive.cpp [hunk 4]
```diff
@@ -363,6 +363,57 @@ char *hijack_grpc_pass(ngx_conf_t *cf, ngx_command_t *command, void *conf) noexc
...
+                      ngx_string("if_not_empty")};
+  ngx_array_t args_array;
+  args_array.elts = static_cast<void *>(&args);
+  args_array.nelts = 4;
+
+  cf->args = &args_array;
+  const auto guard = defer([&]() { cf->args = old_args; });
+
+  for (const std::string_view key : keys) {
+    // NOTE(@dmehala): uWSGI uses the same key header convention thant fastcgi
```
[the remark this answers is not in the document]
Comment (dmehala) [reply src/datadog_directive.cpp]: Brainfart 🤡 Thank you! Resolved in d475f16d67b2f57a20161586b2ddf417903afeaf

File: src/datadog_directive.cpp [hunk 5]
```diff
@@ -363,6 +363,57 @@ char *hijack_grpc_pass(ngx_conf_t *cf, ngx_command_t *command, void *conf) noexc
...
+    args[2] = make_propagation_header_variable(cf->pool, key);
+    auto rcode = datadog_conf_handler({.conf = cf, .skip_this_module = true});
+    if (rcode != NGX_OK) {
+      return static_cast<char *>(NGX_CONF_ERROR);
+    }
+  }
+
+  return static_cast<char *>(NGX_CONF_OK);
+} catch (const std::exception &e) {
+  ngx_log_error(NGX_LOG_ERR, cf->log, 0, "datadog_grpc_propagate_context failed: %s", e.what());
```
[the remark this answers is not in the document]
Comment (dmehala) [reply src/datadog_directive.cpp]: Good catch. Fixed in d475f16d67b2f57a20161586b2ddf417903afeaf

**Review by dgoffredo — APPROVED** [review]
Perfect.

Please see https://github.com/DataDog/nginx-datadog/pull/42 for some documentation tweaks.

File: .gitignore [hunk 6]
```diff
@@ -18,3 +18,6 @@
 
 # scratch directory for preparing releases
 /.release/
+
+# Python
+__pycache__
```
Comment (dmehala) [inline .gitignore:23]:
```suggestion
__pycache__/
```

**Pull request merged**